tools/Microsoft.TemplateEngine.Authoring.Tasks/Tasks/ValidateTemplates.cs

```diff
@@ -0,0 +1,174 @@
+            }
+
+            static string GetTemplateDisplayName(IScanTemplateInfo template)
+            {
+                string templateName = string.IsNullOrEmpty(template.Name) ? "<no name>" : template.Name;
+                return $"'{templateName}' ({template.Identity})";
+            }
+
+            void LogValidationEntries(string subCategory, IReadOnlyList<IValidationEntry> errors)
+            {
+                foreach (IValidationEntry error in errors.OrderByDescending(e => e.Severity))
+                {
+                    switch (error.Severity)
+                    {
+                        case IValidationEntry.SeverityLevel.Error:
+                            Log.LogError(
+                                subcategory: subCategory,
+                                errorCode: error.Code,
+                                helpKeyword: string.Empty,
+                                file: error.Location?.Filename ?? string.Empty,
+                                lineNumber: error.Location?.LineNumber ?? 0,
+                                columnNumber: error.Location?.Position ?? 0,
+                                endLineNumber: 0,
+                                endColumnNumber: 0,
+                                message: error.ErrorMessage);
+                            break;
+                        case IValidationEntry.SeverityLevel.Warning:
+                            Log.LogWarning(
+                                subcategory: subCategory,
+                                warningCode: error.Code,
+                                helpKeyword: string.Empty,
+                                file: error.Location?.Filename ?? string.Empty,
+                                lineNumber: error.Location?.LineNumber ?? 0,
+                                columnNumber: error.Location?.Position ?? 0,
+                                endLineNumber: 0,
+                                endColumnNumber: 0,
+                                message: error.ErrorMessage);
+                            break;
+                        case IValidationEntry.SeverityLevel.Info:
+                            Log.LogMessage(
+                                subcategory: subCategory,
+                                code: error.Code,
+                                helpKeyword: string.Empty,
+                                file: error.Location?.Filename ?? string.Empty,
+                                lineNumber: error.Location?.LineNumber ?? 0,
+                                columnNumber: error.Location?.Position ?? 0,
+                                endLineNumber: 0,
+                                endColumnNumber: 0,
+                                MessageImportance.High,
+                                message: error.ErrorMessage);
+                            break;
+                    }
+                }
+            }
+
+        }
+
+        private CancellationTokenSource GetOrCreateCancellationTokenSource()
```

Comment:
Is this optimization worth the complexity? The cts will need to be created anyways (the only exception is when the `Execute` would not be called at all or if the `TemplateLocation` would be empty - which it should not as it's marked as required).

If the optimization still feels as worth it - should we use `Lazy` instead?

Reply:
I took `LocalizeTemplates` as example here, but I agree. Simplified the logic